src/pudl/transform/ferc1.py

```diff
@@ -1576,29 +1589,35 @@ def transform(
     }
     # create an empty ditctionary to fill up through the transform fuctions
     ferc1_transformed_dfs = {}
-
     # for each ferc table,
     for table in ferc1_tfr_classes:
         if table in ferc1_settings.tables:
             logger.info(
                 f"Transforming raw FERC Form 1 dataframe for loading into {table}"
             )
 
-            ferc1_transformed_dfs[table] = ferc1_tfr_classes[table](
-                table_name=table
-            ).execute(
-                raw_dbf=ferc1_dbf_raw_dfs.get(table),
-                raw_xbrl_instant=ferc1_xbrl_raw_dfs.get(table).get("instant", None),
-                raw_xbrl_duration=ferc1_xbrl_raw_dfs.get(table).get("duration", None),
+            ferc1_transformed_dfs[table] = ferc1_tfr_classes[table]().transform(
+                raw_dbf=ferc1_dbf_raw_dfs[table],
+                raw_xbrl_instant=ferc1_xbrl_raw_dfs[table].get(
+                    "instant", pd.DataFrame()
+                ),
+                raw_xbrl_duration=ferc1_xbrl_raw_dfs[table].get(
+                    "duration", pd.DataFrame()
```

Comment:
Just sayin', I literally enforce that there will be an empty dataframe if one of the instant / duration tables is missing.

In `pudl.extract.ferc1.generic_xbrl_extract()`

```py
    # Not every table contains both instant and duration
    # Return empty dataframe if table doesn't exist
    if table_name not in ferc1_meta.tables:
        return pd.DataFrame()
```

And that is called from `pudl.extract.ferc1.extract_xbrl()`:
```py
        for period_type in ["duration", "instant"]:
            ferc1_raw_dfs[pudl_table][period_type] = generic_xbrl_extract(
                ferc1_engine=sa.create_engine(pudl_settings["ferc1_xbrl_db"]),
                ferc1_settings=ferc1_settings,
                table_name=f"{xbrl_table}_{period_type}",
            )
```

Reply:
I can remove it but I don't see the harm in adding this very simple thing here and there.

Reply:
The only thing that bothers me about it is it feels less readable.